FORMS.md

```diff
@@ -0,0 +1,243 @@
+  - In the case of Field, this attribute should be passed to Input instead of the containing `fieldset` to create the ideal behavior here.
+
+#### Input-specific HTML attributes
+
+Inputs surface a huge list of attributes; these are the most relevant and global.
+
+- `autocomplete`: The autocomplete attribute takes as its value a space-separated string that describes what, if any, type of autocomplete functionality the Input should provide. In cases where we are collecting information about a user, WCAG requires specifying this attribute to set the Field's purpose.
+- `disabled`: Whether the Input is disabled. Disabled Inputs do not receive click events and are not submitted with the Form. See above.
+- `form`: Associates the Input with a form element. See above.
+- `name`: Name of the Input. Submitted with the form as part of a name/value pair.
+- `readonly`: Value is not editable.
+- `required`: A value is required.
+- `type`: Type of Input.
+- `value`: Initial value of the Input.
+
+There are specific attributes that will need to be passed to individual Input types as well. <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input">This chart from MDN</a> provides an excellent breakdown.
+
+### States
+
+Many of these states are assigned or applicable to the `<input>` element. However, those Input states may be better reflected at the Field level. For example: dimming Field.Label for `disabled` elements; showing Field.Error only when an Input is `invalid`.
+
+- `active`: When the Input is activated by the user.
+- `autofill`: When an Input has been auto-filled by the browser.
+- `blank`: When an Input is empty.
+- `checked`: When Inputs like Radio and Checkbox are checked.
+- `dir`: The reading direction currently applied.
+- `disabled`: When an Input is disabled.
+- `enabled`: When an Input is enabled (rather than disabled). This is the default state.
+- `focus-within` When a child has `focus`.
+- `focus`: When the Input has focus.
+- `hover`: When the Input is hovered by a pointing device.
+- `in-range` When an Input is within the range constrictions. e.g. a slider.
+- `indeterminate`: When Inputs are in an indeterminate state. e.g. A "parent" checkbox for a group of mixed checked/unchecked Inputs.
+- `invalid`: When an Input has invalid contents/values.
+- `optional`: When an Input is optional.
+- `out-of-range`: When an Input is outside of range constrictions.
+- `placeholder-shown`: When placeholder text is visible.
+- `read-only`: When an Input cannot be changed by the user.
+- `read-write`: When an Input is user-editable (rather than read-only). This is the default state.
+- `required`: When an Input is required.
+- `target-within`: When a child is a `target`.
+- `target`: When the component is the target of a URL. e.g. `site.com/form#firstname`
+- `user-invalid`: When an Input is invalid, but only after a user has interacted with it.
+- `valid`: When an Input has valid contents/values.
```

Comment:
These are super interesting, today we don't support all of these states. Should we narrow it down a bit to what we plan on supporting?

Reply:
@joejezowski-okta is going to review the state list and determine which ones we need affordances for. I'm not sure we'll need to expand much further than what we have today, but wanted to make sure we ticked the boxes.